libkpod/image/copy.go

```diff
@@ -134,7 +134,19 @@ func PullImage(store storage.Store, imgName string, allTags, quiet bool, sc *typ
 		}
 		// to pull all the images stored in one tar file
 		for i := range manifest {
-			images = append(images, manifest[i].RepoTags[0])
+			if manifest[i].RepoTags != nil {
+				images = append(images, manifest[i].RepoTags[0])
+			} else {
+				// create an image object and use the hex value of the digest as the image ID
+				// for parsing the store reference
+				newImg, err := srcRef.NewImage(sc)
+				if err != nil {
+					return err
+				}
+				defer newImg.Close()
+				hex := newImg.ConfigInfo().Digest.Hex()
```

Comment:
`ConfigInfo()` can return an empty structure, so we should check that the `Digest` field in the returned `BlobInfo` returns `nil` from its `Validate()` method before attempting to read its `Hex()` value.